src/core/lib/json/json_object_loader.h

```diff
@@ -315,38 +328,46 @@ class AutoLoader<Json::Object> final : public LoadUnprocessedJsonObject {};
 template <typename T>
 class AutoLoader<std::vector<T>> final : public LoadVector {
  private:
-  void LoadOne(const Json& json, const JsonArgs& args, void* dst,
-               ErrorList* errors) const final {
+  void* EmplaceBack(void* dst) const final {
     auto* vec = static_cast<std::vector<T>*>(dst);
-    T value{};
-    LoaderForType<T>()->LoadInto(json, args, &value, errors);
-    vec->push_back(std::move(value));
+    vec->emplace_back();
+    return &vec->back();
+  }
+  const LoaderInterface* ElementLoader() const final {
+    return LoaderForType<T>();
   }
 };
 
+template <>
```

Comment:
Please add a comment here explaining that we special-case `vector<bool>` because the underlying `std::vector<>` implementation treats it specially (implements it as a bitmask), so the generic code doesn't work for that case.

Reply:
done